HopsanGenerator/src/generators/HopsanFMIGenerator.cpp

```diff
@@ -1310,6 +1310,29 @@ bool HopsanFMIGenerator::generateToFmu(QString savePath, ComponentSystem *pSyste
     fmuHopsanSourceCode.replace("<<<setdataptrs>>>", setDataPtrsString);
     fmuHopsanSourceCode.replace("<<<timestep>>>", QString::number(pSystem->getDesiredTimeStep()));
 
+    QList<ParameterSpecification> parSpecs;
+    getParameters(parSpecs, pSystem);
+    QString addParametersToMap;
+    for(const auto &parSpec : parSpecs) {
+        QString dataType;
+        if(parSpec.type == "Real") {
+            addParametersToMap.append(QString("        realParametersMap.insert(std::pair<int,hopsan::HString>(%1,\"%2\"));\n").arg(vr).arg(parSpec.name.toStdString().c_str()));
+        }
+        else if(parSpec.type == "Integer") {
+            addParametersToMap.append(QString("        intParametersMap.insert(std::pair<int,std::string>(%1,\"%2\"));\n").arg(vr).arg(parSpec.name));
```

Comment:
I found the compilation error cause. Here and the other non real maps, uses std::string instead of hstring. But the maps are declared using hstring.  I tested to export the unit test model.

Reply:
Could add one of each system parameter type to the fmi export test model

Reply:
Done (except conditional and text block, which are not supported as system parameters).

Reply:
And I also believe I fixed the compilation issues. I can't test it myself though.

Reply:
Yes compilation issue fixed, regarding textblock and conditional, its just a GUI issue, you can actually have them if you edit the type in the hmf file manually. :) I forgot to fix that.